lib/gitWorkflow.js

```diff
@@ -109,16 +93,29 @@ class GitWorkflow {
       // Manually check and backup if necessary
       await this.backupMergeStatus()
 
+      // Get a list of unstaged deleted files, because certain bugs might cause them to reappear:
+      // - in git versions =< 2.13.0 the `--keep-index` flag resurrects deleted files
+      // - git stash can't infer RD or MD states correctly, and will lose the deletion
+      this.deletedFiles = (await this.execGit(['ls-files', '--deleted']))
+        .split('\n')
+        .map(file => path.resolve(this.gitDir, file))
+
       // Save stash of entire original state, including unstaged and untracked changes.
       // `--keep-index leaves only staged files on disk, for tasks.`
       await this.execGit(['stash', 'save', '--include-untracked', '--keep-index', STASH])
 
-      // Restore meta information about ongoing git merge
-      await this.restoreMergeStatus()
-
       // There is a bug in git =< 2.13.0 where `--keep-index` resurrects deleted files.
       // These files should be listed and deleted before proceeding.
-      await cleanUntrackedFiles(this.execGit)
+      const untrackedFiles = (await execGit(['ls-files', '--others', '--exclude-standard']))
+        .split('\n')
+        .map(file => path.resolve(this.gitDir, file))
+
+      for (const file of untrackedFiles) {
+        await removeFile(file)
+      }
```

Comment:
How about, instead,
```js
await Promise.all(untrackedFiles.map(file => removeFile(file)))
```
to run all the deletions at once instead of in sequence?

Reply:
Thanks for the suggestion. I figured it wouldn't matter perf-wise, but it looks clean anyway.